### ISSUE TITLE
Mactype loading mode poll, please voice your opinions

### ISSUE THREAD

**Issue by snowie2000** [snowie2000/mactype#729]
Please vote for your loading mode, thanks

[![](https://api.gh-polls.com/poll/01F0WEJABRPR918KSYZKAN2BJX/Tray%20mode)](https://api.gh-polls.com/poll/01F0WEJABRPR918KSYZKAN2BJX/Tray%20mode/vote)
[![](https://api.gh-polls.com/poll/01F0WEJABRPR918KSYZKAN2BJX/Service%20mode)](https://api.gh-polls.com/poll/01F0WEJABRPR918KSYZKAN2BJX/Service%20mode/vote)
[![](https://api.gh-polls.com/poll/01F0WEJABRPR918KSYZKAN2BJX/Registry%20mode)](https://api.gh-polls.com/poll/01F0WEJABRPR918KSYZKAN2BJX/Registry%20mode/vote)
[![](https://api.gh-polls.com/poll/01F0WEJABRPR918KSYZKAN2BJX/Manual%20mode)](https://api.gh-polls.com/poll/01F0WEJABRPR918KSYZKAN2BJX/Manual%20mode/vote)

-----------------------------

Also, please kindly vote if you agree to retire the registry mode or not since it's getting deprecated by Microsoft and not effective with secure boot on.
[![](https://api.gh-polls.com/poll/01F0WEMMW843WT73K5HNBQCEXH/Agree%20to%20retire%20the%20registry%20mode)](https://api.gh-polls.com/poll/01F0WEMMW843WT73K5HNBQCEXH/Agree%20to%20retire%20the%20registry%20mode/vote)
[![](https://api.gh-polls.com/poll/01F0WEMMW843WT73K5HNBQCEXH/Please%20keep%20it%20no%20matter%20how%20buggy%20it%20is)](https://api.gh-polls.com/poll/01F0WEMMW843WT73K5HNBQCEXH/Please%20keep%20it%20no%20matter%20how%20buggy%20it%20is/vote)

**Comment by sammilucia** [comment]
I would add that it may be useful to keep registry mode active for earlier versions of windows if people are still using them. though Service and Tray Mode work so well there may be no need

**Comment by snowie2000** [comment]
Registry mode is getting bad for newer Windows versions. For old versions before Windows 10, they are acceptable.

Unfortunately, the poll can't be edited. I can't add another entry to it.

**Comment by rpzone** [comment]
With the development of windows 10, registry mode brings more and more troubles in daily use. Strongly recommend to update mactype again and make it more compitable and stable.

**Comment by AhmedAdelElazab** [comment]
I still using registry mode on windows 10 20H2 (build 19042.867) with secure boot disabled in uefi and i had no problems at all.

**Comment by sammilucia** [comment]
here's a question... does Registry Mode offer anything that Tray Mode (Administrator) or Service Mode _doesn't_?

if it doesn't ... why even have it if it's causing problems?

**Comment by snowie2000** [comment]
In theory, it's a more "natural" way to inject DLLs.
No external thread is created for remote processes, which is sometimes anti-malware friendly.

The time of loading in registry mode is also in quite an early stage. The registry mode injection is done by Windows `gdi32.dll` so when this system DLL is being loaded, so is MacType. And because of the way it works, command-line applications that do not rely on gdi32.dll won't load MacType naturally (in case the `hookChildProcess` option is not set, which, someone does do).

And because Windows is doing this for us, there is no need to polling the process list or watching for process creations. Conserving sort of "considerable" CPU usage. It is unnoticeable on my system, but there are reports that it may peak up to 10%.

**Comment by ssssssbbb** [comment]
I don't think it's necessary to remove the entry of registry mode unless it causes boot failure which is hard to be resumed, it's just application error for now.

U can take the easy way, maybe adding 3 repeating confirm dialogs with beeping or barking is much easier for programming.

**Comment by snowie2000** [comment]
It's actually causing boot failures for some system. And although it can be disabled in the safe mode, not everyone knows the way to enter safe mode under windows 10 boot manager. (F8 during startup no longer worked)

Confirmation dialogs mean nothing to users. They will simply yes to all and complain afterwards.

**Comment by sammilucia** [comment]
yeah, I think you're right @snowie2000. MacType already has some reputation for being unreliable even though usually it's a problem with antivirus locking files or games and applications doing specific weird things with fonts. although a lot of that stuff is not actually MacType (which injects politely into software), I think we should still be addressing the perception of 'reliability'.

and you're right, a lot of users will just click accept. the current version of MacType already has a warning about registry mode in the description and people still use it and have problems.

it has to go.

I think instead we can focus on a master exclusion list of software that's known to have problems (exes without UIs, several games, etc.) which could reduce the amount of apps MacType has to scan and possibly improve performance.

although <1% CPU isn't much these days. normally dwm and other windows processes are using more than MacType.

**Comment by ssssssbbb** [comment]
Well, maybe I underestimated the reg problem, especially with Bug10.

New proplem is hardware related, as for exclusion list I have hunderds of exes on it but that's not the point, I have no problems with that kind of special exes for 9 or 10 years. I stoped updating the list and considered it as a list for exes which can't be rendered well but no error.

**Comment by snowie2000** [comment]
> Well, maybe I underestimated the reg problem, especially with Bug10.
> 
> 
> 
> New proplem is hardware related, as for exclusion list I have hunderds of exes on it but that's not the point, I have no problems with that kind of special exes for 9 or 10 years. I stoped updating the list and considered it as a list for exes which can't be rendered well but no error.

The CET compatibility problem has partially been solved and there is patch for it in https://github.com/snowie2000/mactype/issues/731#issuecomment-816621955

An x86 fix will be available once I have time on it.

**Comment by matoi974** [comment]
Some people reported that newer Firefox versions require Registry mode for Mactype to work, though in my experience it didn't work either way, couldn't figure out why. If registry mode is indeed the only way to hook new Firefox currently, that's bad news for Firefox users if it's going to stop working in the future.
I have to use Tray mode because I need to be able to disable Mactype without rebooting for applications like Mod Organizer to work properly, I don't really have a choice here.

**Comment by xSwagboy69x** [comment]
I'm surprised that 'Service Mode' seems to be most used here; am I the only one with issues like 'partial rendering' on service mode? For example:

![2021_05_09 - 04 47 00 PM](https://user-images.githubusercontent.com/13977077/117586559-dd7f3480-b0e6-11eb-9c5f-4b20b0fc98c3.jpg)

I just switched on **Service Mode** to once again test, and it still does this; happens often, across many programs and parts of the UI, usually with dialogue boxes.

In **Registry Mode**, Mactype renders perfectly and without delay, every time. If the other modes could perform like that, I'd switch since you're saying it's better in some ways, but if not I'd prefer Registry Mode to stick around.

Also, @matoi974 : Why can't you use MO with MacType on? I use MO all the time and never bothered to switch it off, it renders the program fine, at least on the surface. Haven't noticed anything weird.

![2021_05_06 - 10 07 00 AM](https://user-images.githubusercontent.com/13977077/117586777-461ae100-b0e8-11eb-977d-fe6c407f58c1.jpg)

**Comment by sammilucia** [comment]
the new version will still have registry mode, it just needs to be enabled manually (not through the Wizard). I think this is a good balance because it's still there, but will help prevent a lot of common problems from people just selecting it (and having problems). we're working on releasing this asap 😊

**Comment by garoto** [comment]
Currently using Win8.1 since 2016, and before that Win7 since 2013 or so and MacType since it was known as `gdi++` (yes, I know not the same project yada yada). 

Settled for using Registry mode from the beggining. Can't think why anyone would choose anything other `Registry Mode` to be quite honest: none of the other modes offer the level of flawless injection as `Appinit_Dlls` does. 

The fact that this mode of injection is considered for 100% deprecation makes me worried and kinda annoyed.

**Comment by sammilucia** [comment]
@garoto please see my comment above 😊

we hear you. we know we have a lot of non-Win10 users who love reg mode. unfortunately MS is pushing Win10 hard and reg mode just fails on most platforms because of tighter Windows security. we tried making it "not recommended" in the Wizard, but we still get more requests for help than we can respond to. we all love MacType but work on it in our spare time.

it will still be there for long-time MacType users 😊

**Comment by garoto** [comment]
@sammilucia: I got you. I guess I'm just confused to what I'm gonna do in the future regarding Windows corporate sponsored spyware and whatnot.

**Comment by sammilucia** [comment]
@garoto yeah I know 🙄. I modify Win10 pretty hard to remove most of it, I have a procedure I made, I can post it somewhere if you want?

**Comment by snowie2000** [comment]
Actually, the registry mode still works even with the latest release (not published yet), you just can't enable it by simply click and go in the wizard. A detailed guide will be provided in the wiki later.

And you are still able to enable registry mode via the wizard if you're on Windows version earlier than windows 10.

**Comment by matoi974** [comment]
@Jessamynn All Skyproc patchers (Requiem's, SkyRe's, etc) break and don't see any .esp's installed. It might have happened because I was using tray mode, but disabling Mactype fixed this for me.

**Comment by xSwagboy69x** [comment]
@sammilucia Cool, sounds like a reasonable compromise :)

@matoi974 Hm, that's odd. I guess I haven't used that kind of patcher since LE; good to know. Good luck!

**Comment by ChicoThorn** [comment]
> here's a question... does Registry Mode offer anything that Tray Mode (Administrator) or Service Mode _doesn't_?

Yes. At least from my experience. — I really WANT to use Service Mode... I really WANT to like using Service Mode... but that whole 'wait to hook' or 'hover over text first' or 'minimize then restore window' to get MacType to kick in drives me batty! That doesn't happen with Registry Mode (well, at least until they started porting Sys32 apps to the Microsoft Store; those things seem to break it pretty badly). I guess I just want Service Mode to behave like I'm used to Registry Mode behaving: rendering fonts & text beautifully without me having to think about it. When I open a window or an app and suddenly the text isn't MacType rendered it distracts me from my work and I want to immediately find a solution to it. I'm not saying continuing on with Registry Mode is the answer, because obviously if Microsoft is deprecating the way it works then eventually it will just break and that'll be it. — I'm hoping that @snowie2000 can wave their magic wand and make Service Mode flawless, lol! That'd satisfy my OCD fixation with this for sure! 😂

**Comment by ChicoThorn** [comment]
> am I the only one with issues like 'partial rendering' on service mode?

I have that same issue with Service mode myself. Up until MacType 2021-rc1 I was a dyed-in-the-wool Registry fan. But with Registry mode being deprecated, not only by MacType, but by its effectiveness in Windows as Microsoft continues to change security measures. Most notably are new stringent requirements for running Windows 11. The computer MUST have Secure Boot enabled or Windows 11 won't run. Registry mode won't work with Secure Boot enabled, so the point becomes moot.  —— After experimenting with both Service mode and Tray mode I find that I like Tray mode much more. For the most part all text throughout the UI renders properly. The only hiccup I've noticed is right after computer startup. After refreshing the window MacType in Tray mode works fine. I also like Tray mode because it puts an icon in the SysTray with many helpful tools and selections available. Most important for me is the ability to 'Restart' MacType on the fly. This is particularly great when fine tuning and testing INI settings.

**Comment by gfody** [comment]
registry mode + service. registry mode works fine mostly and the service handles the UWP apps.

win10 version 21H1 (19043.1586)

**Comment by ChicoThorn** [comment]
> registry mode + service.

@gfody — How do you set it to both Registry Mode and Service Mode? I take it you're running both at the same time?

**Comment by gfody** [comment]
install the service and then manually edit the registry (steps from the wiki)

**Comment by ChicoThorn** [comment]
@gfody — Thanks! It worked! 🙂
